lib/code_corps/services/user_service.ex

```diff
@@ -0,0 +1,99 @@
+defmodule CodeCorps.Services.UserService do
+  @moduledoc """
+  Handles CRUD operations for users.
+
+  When operations happen on `CodeCorps.User`, we need to make sure changes
+  are propagated to related records, ex., `CodeCorps.StripePlatformCustomer` or
+  `CodeCorps.StripeConnectCustomer`
+
+  """
+
+  alias CodeCorps.{Repo, StripeConnectCustomer, StripePlatformCustomer, User}
+  alias CodeCorps.StripeService.{StripeConnectCustomerService,StripePlatformCustomerService}
+  alias Ecto.Changeset
+  alias Ecto.Multi
+
+  @doc """
+  Updates a `CodeCorps.User` record and, if necessary, associated
+  `CodeCorps.StripePlatformCustomer` and `CodeCorps.StripeConnectCustomer` records.
+
+  These related records inherit the email field from the user,
+  so they need to be kept in sync, both locally, and on the Stripe platform.
+
+  Returns one of
+  * `{:ok, %CodeCorps.User{}, :nothing_to_update, :nothing_to_update}`
+  * `{:ok, %CodeCorps.User{}, %CodeCorps.StripePlatformCustomer{}, :nothing_to_update}`
+  * `{:ok, %CodeCorps.User{}, %CodeCorps.StripePlatformCustomer{}, %CodeCorps.StripeConnectCustomer{}}`
+  * `{:error, %Ecto.Changeset{}}`
+  * `{:error, :unhandled}`
+
+  """
+  def update(%User{} = user, attributes) do
+    changeset = user |> User.update_changeset(attributes)
+    do_update(changeset)
+  end
+
+  defp do_update(%Changeset{changes: %{email: _email}} = changeset) do
+    multi =
+      Multi.new
+      |> Multi.update(:update_user, changeset)
+      |> Multi.run(:update_platform_customer, &update_platform_customer/1)
+      |> Multi.run(:update_connect_customers, &update_connect_customers/1)
+
+    case Repo.transaction(multi) do
+      {:ok, %{
+        update_user: user,
+        update_platform_customer: update_platform_customer_result,
+        update_connect_customers: update_connect_customers_results
+      }} ->
+        {:ok, user, update_platform_customer_result, update_connect_customers_results}
+      {:error, :update_user, %Ecto.Changeset{} = changeset, %{}} ->
+        {:error, changeset}
+      {:error, _failed_operation, _failed_value, _changes_so_far} ->
+        {:error, :unhandled}
+      other ->
+        IO.inspect(other, pretty: true)
+    end
+  end
+
+  defp do_update(%Changeset{} = changeset) do
+    with {:ok, user} <- Repo.update(changeset) do
+      {:ok, user, :nothing_to_update, :nothing_to_update}
```

Comment:
`:nothing_to_update` seems overly specific here whereas `nil` is just as descriptive, IMO.